project1-sps/script.js

```diff
@@ -1,4 +1,78 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+// Step 1 : Create mathematical equation for bot to generate a try
+  // Step 1a: Attach a randomNumber to scissors, paper or stone
+
+var botTry = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomTry;
+  var randomNumber = Math.floor(randomDecimal); // scissors, paper, stone
+
+  if (randomNumber == 0) {
+    return "scissors";
+  } 
+  
+  if (randomNumber == 1) {
+     return "paper";
+  }
+  
+  if (randomNumber == 2) {
+    return "stone";
 };
+
+
+// Step 2: Create a functional logic to determine how player wins the game
+  // scissors beats paper not stone
+  // paper beats stone not scissors
+  // stone beats scissors not paper
+
+
+var playerToWin = function (playerTry, botTry) {
+  return (
+    (playerTry == scissors && botTry == paper) ||
+    (playerTry == paper && botTry == stone) ||
+    (playerTry == stone && botTry == scissors)
+  )
+};
+
+
+// Step 3: Create a functional logic to determine if it's a draw
+  // scissors = scissors
+  // paper = paper
+  // stone == stone
+
+var playerToDraw = function (playerTry, botTry) {
+  return (
+    (playerTry == scissors && botTry == scissors) ||
+    (playerTry == paper && botTry == paper) ||
+    (playerTry == stone && botTry == stone)
+  )
+};
+
+
+// Step 4: Create a functional logic to determine if player loses
+  // scissors < stone
+  // paper < scissors
+  // stone < paper
+
+var playerToLose = function (playerTry, botTry) {
+  return (
+    (playerTry == scissors && botTry == stone) ||
+    (playerTry == paper && botTry == scissors) ||
+    (playerTry == stone && botTry == paper)
+  )
+};
+
+
+// Step 5: Define messaging for various outcomes (to Step 2, 3 or 4)
+
+
+if (input == playerToWin) {
+  return "Congrats you WON!";
+};
+
+if (input == playerToDraw) {
+  return "It's a DRAW. Try again.";
+}; 
+
+if (input == playerToLose) {
+  return "Oh no, you LOST"
+}
```

Comment:
Your code is missing a main function now - the only way we can interact with the code is through the submit button, which triggers the main function.
Refer to this as a guide on what to put in your main function:
```
var main = function (input) {
  // Generate a random move for the computer (use the function from your Step 1)
  var botMove = generateRandomMove();

  // Create an empty output message
  var myOutputValue = "";

  // Run your if conditions to edit the message (what you were trying to do in Step 5, but look at how to use your helper functions from steps 2-4 for the if condition)
  if (playerToWin(input, botMove)) {
    myOutputValue = "win...";
  } // and so on... same for draw and lose

  // Send output back to browser
  return myOutputValue;
};
```